Instant search: introduce overlay from right

bluefuton opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to to-test.md in a new commit as part of your PR. -->

This PR changes the Instant Search overlay to appear from the right, rather than from above. 

In #14230 @gibrown commented:

> For some reason I want it to slide up or from the side that the filters are on. More like turning a page...

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?

No - this is part of the Instant Search prototype.

#### Testing instructions:

Follow the setup instructions in https://github.com/Automattic/jetpack/blob/instant-search-master/modules/search/instant-search/README.md.

#### Proposed changelog entry for your changes:

Not required.

bluefuton committed:
Introduce overlay from right

gibrown commented:
I like it better. Makes me want to slide in search results when I start typing too ;)

keoshi committed:
Updating overlay offset values

- Changes units from `vh` to `vw`.
- Updates value from `200` to `100`, as [mentioned here](https://github.com/Automattic/jetpack/pull/14320#discussion_r366365031).

keoshi commented:
Pushed a small change and tested this on:

- Chrome Version 79.0.3945.88 (Official Build) (64-bit)
- Firefox 71.0 (64-bit) and 72.0.1 (64-bit)
- Safari Version 13.0.4 (15608.4.9.1.3)

keoshi reviewed (approved):
Looks good and I much prefer the slide-from-the-right transition as mentioned previously.

Pushed a small change in units and value, but it should be good to go after another quick review.

bluefuton commented:
Thanks @keoshi 🙂
